x-pack/plugins/maps/public/store/map.js

```diff
@@ -359,3 +361,60 @@ function getValidDataRequest(state, action, checkRequestToken = true) {
 function findLayerById(state, id) {
   return state.layerList.find(layer => layer.id === id);
 }
+
+function rollbackAnyTrailingTrackedLayerState(state) {
+  let mutatedState = state;
+  state.layerList.forEach(layer => {
+    if (layer[TRACKED_LAYER_DESCRIPTOR]) {
+      mutatedState = rollbackTrackedLayerState(state, layer.id);
```

Comment:
This implementation is going to cause problems. `mutatedState` will end up with only the last item in LayerList with tracked state removed because the input parameter `state` is not getting updated. 

Is there ever a case where you would want `ROLLBACK_TO_TRACKED_LAYER_STATE ` to not also do `ROLLBACK_ANY_TRAILING_TRACKER_LAYER_STATE `?

I would recommend removing `ROLLBACK_TO_TRACKED_LAYER_STATE` action and just having a single action `ROLLBACK_ANY_TRAILING_TRACKER_LAYER_STATE`. Then `removeTrackedLayerState` can just return the updated layer instead of the entire state and this method will be much simpler because it can just create a new layerList in the forEach.